spec/01-unit/27-queue_spec.lua

```diff
@@ -392,6 +416,128 @@ describe("plugin queue", function()
     )
   end)
 
+  it("works when time is moved forward while items are being queued", function()
+    local number_of_entries = 1000
+    local number_of_entries_processed = 0
+    now_offset = 1
+    for i = 1, number_of_entries do
+      Queue.enqueue(
+        queue_conf({
+          name = "time-forwards-adjust",
+          max_batch_size = 10,
+          max_coalescing_delay = 10,
+        }),
+        function(_, entries)
+          number_of_entries_processed = number_of_entries_processed + #entries
+          return true
+        end,
+        nil,
+        i
+      )
+      now_offset = now_offset + now_offset
+    end
+    helpers.wait_until(
+      function ()
+        return number_of_entries_processed == number_of_entries
+      end,
+      10
+    )
+  end)
+
+  it("works when time is moved backward while items are being queued", function()
+    local number_of_entries = 1000
+    local number_of_entries_processed = 0
+    now_offset = -1
+    for i = 1, number_of_entries do
+      Queue.enqueue(
+        queue_conf({
+          name = "time-backwards-adjust",
+          max_batch_size = 10,
+          max_coalescing_delay = 10,
+        }),
+        function(_, entries)
+          number_of_entries_processed = number_of_entries_processed + #entries
+          return true
+        end,
+        nil,
+        i
+      )
+      now_offset = now_offset + now_offset
+    end
+    helpers.wait_until(
+      function ()
+        return number_of_entries_processed == number_of_entries
+      end,
+      10
+    )
+  end)
+
+  it("works when time is moved backward while items are on the queue and not yet processed", function()
```

Comment:
I'd appreciate a little explanation of what's happening here.

Reply:
Comments added.  Also a couple of sleeps while sending to cause the process to yield.